src/core/lib/iomgr/fork_posix.cc

```diff
@@ -48,6 +49,7 @@ bool registered_handlers = false;
 
 void grpc_prefork() {
   skipped_handler = true;
+  grpc_maybe_wait_for_async_shutdown();
```

Comment:
I don't think this is necessary here (if it is, it needs to go after the check of `grpc_is_initialized()` below, as this may be called by the OS after core has shutdown). 

Apologies for the lack of test coverage around this; still working through some unrelated issues, but https://github.com/grpc/grpc/pull/16513 is adding tests for the fork use case.

Core does not shutdown as part of its pre-fork handlers; Python (and PHP, as far as I know) shut down core in the child's post-fork handler, but that is handled outside of core's fork handlers.